<?xml version="1.0"?>
<launch>
  <!-- ZED2i Debug Obstacle Detector Launch File -->
  <!-- For development and testing with debug output enabled -->
  
  <node name="zed_obstacle_detector" pkg="zed_obstacle_detector" type="zed_obstacle_detector_node" output="screen">
    
    <!-- Input/Output Topics -->
    <param name="point_cloud_topic" value="/zed2i/zed_node/point_cloud/cloud_registered" />
    
    <!-- Frame Configuration -->
    <param name="base_link_frame" value="base_link" />
    <param name="world_frame" value="world" />
    <param name="camera_frame" value="zed2i_left_camera_frame" />
    
    <!-- Processing Parameters (Debug settings) -->
    <param name="passthrough_z_min_camera" value="0.2" />
    <param name="passthrough_z_max_camera" value="7.0" />
    <param name="voxel_leaf_size" value="0.05" />  <!-- Finer resolution for debug -->
    <param name="enable_uniform_downsampling" value="false" />
    <param name="uniform_sampling_radius" value="0.05" />
    <param name="enable_early_exit" value="false" />  <!-- Disable for debug -->
    <param name="min_points_for_processing" value="100" />
    <param name="max_points_for_processing" value="20000" />
    
    <!-- Ground Detection (Debug settings) -->
    <param name="enable_ground_filtering" value="true" />
    <param name="ground_filter_distance_threshold" value="0.05" />
    <param name="ground_filter_angle_threshold_deg" value="10.0" />
    <param name="ground_filter_max_iterations" value="150" />
    <param name="mars_terrain_mode" value="false" />
    
    <!-- Clustering (Debug settings) -->
    <param name="cluster_tolerance" value="0.25" />
    <param name="min_cluster_size" value="15" />
    <param name="max_cluster_size" value="10000" />
    
    <!-- Transform Parameters (Debug settings) -->
    <param name="tf_lookup_timeout" value="0.1" />
    <param name="tf_buffer_duration" value="10.0" />
    
    <!-- Tracking Parameters (Debug settings) -->
    <param name="obstacle_association_distance" value="2.0" />
    <param name="obstacle_timeout_sec" value="60.0" />
    <param name="position_smoothing_factor" value="0.3" />
    <param name="min_detections_for_confirmation" value="2" />
    
    <!-- Performance Monitoring (Debug settings) -->
    <param name="enable_detailed_timing" value="true" />  <!-- Enable for debug -->
    <param name="enable_debug_publishers" value="true" />  <!-- Enable for debug -->
    <param name="timing_report_interval" value="10.0" />
    <param name="enable_performance_logging" value="true" />
    <param name="log_file_path" value="/tmp/zed_obstacle_detector_debug.log" />
    
    <!-- Debug Output -->
    <param name="enable_debug_output" value="true" />
    
  </node>
  
  <!-- RViz for visualization -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find zed_obstacle_detector)/config/zed_obstacle_detector.rviz" />
  
</launch> 